app/models/account/credit_card.rb

```diff
@@ -96,8 +96,12 @@ def delete_cc_details
   end
 
   def unstore_credit_card!
-    response = provider_payment_gateway.try!(:threescale_unstore, credit_card_auth_code)
-    log_gateway_response(response, "unstore [auth: #{credit_card_auth_code}]")
+    begin
```

Comment:
this line can be removed

Reply:
So, I trusted you and removed the line, but then I realized that this `end` in line `:104` doesn't actually end the method, so I think we can't get rid of `begin`.

Reply:
my bad, I overlooked that, thought you added by mistake

Reply:
Although then there is a check `return if payment_detail.destroyed?` which idk how it is supposed to work. Maybe we should make the rescue global for the method.

Reply:
I think these are need to be executed in any case:
```
    self.credit_card_auth_code = nil
    self.credit_card_expires_on = nil
    self.credit_card_partial_number = nil
```

Reply:
I wouldn't care about these other values. If things work in real life as they are now, then I'm fine with it. I was just worried that by invalid configuration `payment_detail` may raise. But I think your test ensures it wouldn't.